test/unit/forge/ee/routes/api/pipeline_spec.js

```diff
@@ -0,0 +1,484 @@
+                    payload: {
+                        name: 'stage-two',
+                        instanceId: TestObjects.instanceOne.hashid // in use
+                    },
+                    cookies: { sid: TestObjects.tokens.alice }
+                })
+
+                const body = await response.json()
+
+                body.should.have.property('code', 'unexpected_error')
+                body.should.have.property('error').match(/instanceId/)
+
+                response.statusCode.should.equal(500)
+            })
+        })
+    })
+
+    describe('Get Pipeline Stage', function () {
+        it('Should return a single pipeline stage', async function () {
+            const pipelineId = TestObjects.pipeline.hashid
+            const stageId = TestObjects.stageOne.hashid
+
+            const response = await app.inject({
+                method: 'GET',
+                url: `/api/v1/pipelines/${pipelineId}/stages/${stageId}`,
+                cookies: { sid: TestObjects.tokens.alice }
+            })
+
+            const body = await response.json()
+
+            body.should.have.property('id')
+            body.should.have.property('name', 'stage-one')
+            body.should.have.property('instances')
+            body.instances[0].should.have.property('name', 'project1')
+
+            response.statusCode.should.equal(200)
+        })
+    })
+
+    describe('Update Pipeline Stage', function () {
+        describe('With a new name', function () {
+            it('Should update a single pipeline stage with a new name', async function () {
+                const pipelineId = TestObjects.pipeline.hashid
+                const stageId = TestObjects.stageOne.hashid
+
+                const response = await app.inject({
+                    method: 'PUT',
+                    url: `/api/v1/pipelines/${pipelineId}/stages/${stageId}`,
+                    payload: {
+                        name: 'New Name'
+                    },
+                    cookies: { sid: TestObjects.tokens.alice }
+
+                })
+
+                const body = await response.json()
+
+                body.should.have.property('id')
+                body.should.have.property('name', 'New Name')
+                body.should.have.property('instances')
+                body.instances[0].should.have.property('name', 'project1')
+
+                response.statusCode.should.equal(200)
+            })
+        })
+
+        describe('With a new instance', function () {
+            it('Should unassign the old instance and assign the new one', async function () {
+                const pipelineId = TestObjects.pipeline.hashid
+                const stageId = TestObjects.stageOne.hashid
+
+                const response = await app.inject({
+                    method: 'PUT',
+                    url: `/api/v1/pipelines/${pipelineId}/stages/${stageId}`,
+                    payload: {
+                        instanceId: TestObjects.instanceTwo.id
```

Comment:
Need to check what `TestObjects.instanceTwo.id` is, I expect it needs to be a UUID not a hashid.

Will try and dig later.